internal/cli/cli.go

```diff
@@ -132,6 +135,15 @@ func (c *banzaiCli) Client() *pipeline.APIClient {
 		config.HTTPClient.Transport.(*oauth2.Transport).Base = c.RoundTripper()
 
 		c.client = pipeline.NewAPIClient(config)
+
+		resp, _, err := c.client.CommonApi.ApiVersionGet(context.Background())
+		if err != nil {
+			LogAPIError("query API version", err, nil)
+			log.Fatalf("failed to query API version: %v", err)
+		}
+		if ver := resp.Version; ver != supportedPipelineAPIVersion {
```

Comment:
exact match is a no-go for me

Reply:
https://pkg.go.dev/github.com/Masterminds/semver/v3?tab=doc#Version.Compare

Reply:
Yeah, I know that we should check semver, but what would the semantics be?

Reply:
we know the first version that is compatible with this cli code. we should check if the server version is equal or greater than it.

Reply:
But a newer server version might not be compatible with this CLI code.